packages/abi/src/coder/abi-coder.ts

```diff
@@ -1,2 +1,11 @@
+import { getCoder } from './encoding/v1/coder-matcher';
+import type { Coder } from './encoding/v1/coders/types';
+
 // Placeholder
-export class AbiCoder {}
+export class AbiCoder {
+  public encode(type: string, value: unknown) {
+    const coder = getCoder(type) as Coder & { encode: (value: unknown) => unknown };
```

Comment:
Please do not construct types on the fly.

Have types clearly defined in a way they can be imported and used.

Reply:
I agree and I wouldn't do it if we had decided on what the `Coder` interface would look like. This is all a work in progress to show how it'd look like to get the coder and call a fictitious `encode` method "in action". The point of this PR is to showcase how we'd go from a sway type to a coder or typer, whatever they look like in the end. Subsequent PRs will introduce coders and typers properly.

Reply:
After a discussion with @petertonysmith94, I removed the coder and typer examples out of the code and into the PR description.